File: dsa-interview-crash-course/643_max_subarray_i.py
from typing import List


class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        subSum = 0
        for i in range(k):
            subSum += nums[i]
        
        maxAvg = subSum / k
        for i in range(k, len(nums)):
            subSum -= nums[i-k]
            subSum += nums[i]
            maxAvg = max(maxAvg, subSum/k)
        
        return maxAvg

File: dsa-interview-crash-course/643_max_subarray_i.cpp
#include <vector>

class Solution
{
public:
    double findMaxAverage(std::vector<int> &nums, int k)
    {
        double subSum = 0;
        for (int i = 0; i < k; ++i)
            subSum += static_cast<double>(nums[i]);
        double maxAvg = subSum / k;
        for (int i = k; i < nums.size(); ++i)
        {
            subSum -= nums[i - k];
            subSum += nums[i];
            maxAvg = std::max(maxAvg, subSum / k);
        }
        return maxAvg;
    }
};